packages/core/src/app/customer/CheckoutButtonContainer.tsx

```diff
@@ -113,7 +105,7 @@ function mapToCheckoutButtonContainerProps({
     checkoutService,
 }: CheckoutContextProps): WithCheckoutCheckoutButtonContainerProps | null {
     const config = getConfig();
-    const availableMethodIds = filterUnsupportedMethodIds(config?.checkoutSettings.remoteCheckoutProviders ?? []);
+    const availableMethodIds = getSupportedMethodIds(config?.checkoutSettings.remoteCheckoutProviders ?? []);
```

Comment:
Can we limit this call to only V1 buttons? V2 button shouldn't need it at all, as `remoteCheckoutProviders` that don't have a button won't resolve with anything. The reason we want to limit to V1 only is because we don't want to encourage teams to add more IDs to the list of "supported methods".

Reply:
Yes that's correct in a future state when all buttons have been migrated to their own packages, but right we need to pass a list of methodIds to V2 button component to maintain parity and deprecate V1 list.

Once all buttons have been migrated to their respective packages we can get rid of the methods that need to be passed to the component.

Reply:
I meant move the call so that the result of the call wouldn't affect V2 buttons.

I notice that you need the count of `availableMethodIds` in order to set the correct CSS class for the container (e.g.: `checkout-buttons--1`). This means we will still need to modify `SUPPORTED_METHODS` in the `core` package every time we add a new V2 button.

To solve this problem, I think we need to resolve the V2/V1 buttons earlier. For example, in `CheckoutButtonContainer`, we could have:

```tsx
const resolvedCheckoutButtons = remoteCheckoutProviders.reduce((buttons, methodId) => {
    const resolvedCheckoutButtonV2 = resolveCheckoutButtonV2({ id: methodId });

    if (resolvedCheckoutButton) {
        return [...buttons, resolvedCheckoutButtonV2];
    }

    const resolvedCheckoutButtonV1 = resolveCheckoutButtonV1({ id: methodId });

    if (resolvedCheckoutButtonV1) {
        return [...buttons, resolvedCheckoutButtonV2];
    }

    return buttons;
}, []);
```

And in `resolveCheckoutButtonV1`, we could check to see if `methodId` is in the list of `SUPPORTED_METHODS`.

That way, `resolvedCheckoutButtons.length` can give you the correct count for the container while the buttons are loading. And `SUPPORTED_METHODS` only affects V1 button.

Reply:
Right now we have the `SUPPORTED_METHODS` in other places as well. So I suppose we can move the buttons V2 component to accept a list of resolved buttons(combination of V1 and V2) as props. And they can be rendered accordingly.

Reply:
Wanted to check do we want to remove V2 List component, if we are resolving buttons in Container itself. I say so since we might need to pass more information to V2 list in this case such as is the button a V1 or V2 component since they have different interfaces. So might need to pass an object with 
`[ { type: 'V1', methodId: 'applepay', component: V1Component }, { type: 'V2', methodId: 'googlepay', component: V2Component } ]`?